lib/plugin.js

```diff
@@ -104,7 +105,16 @@ export default <%= options.blockLoading ? 'async ' : ''%>(context, inject) => {
     // we might not know the to's page title in vue-router.afterEach, DOM is updated _after_ afterEach
     tracker.setDocumentTitle(to.path)
     <% } %>
-    tracker.setCustomUrl(baseUrl + to.fullPath)
+
+    <% if(options.trailingSlash) { %>
+      const fullPath = withTrailingSlash(to.fullPath)
```

Comment:
This doesnt do anything when the url contains a query or hash, eg for the fullPath `/product?page=4` the trailng slash will just be added at the end (`/product?page=4/`) which is wrong ofc. Easiest way to fix this is probably to use `to.path` here and then after fixing the slash append `to.fullPath.slice(to.path.length )`

Reply:
> Hey, thanks for looking into this.
> 
> Was wondering why we wouldnt just require the user to set the nuxt config option [`trailingSlash`](https://nuxtjs.org/docs/2.x/configuration-glossary/configuration-router#trailingslash) for this? Cause we can fix it in Matomo reporting, but then the canonical url for seo would still be wrong.

Using the router config would be the easiest way. For static sites (hosted on netlify for example) we would have to rely on proper redirects. Currently i don't think it's possible with netlify to redirect from `example.com/a/` to `example.com/a`, only [the other way around](https://docs.netlify.com/routing/redirects/redirect-options/#trailing-slash).

We can close this anyway because it seems a bit too hacky.